docs/src/user/basic_setup.md

```diff
@@ -6,16 +6,35 @@ Youki currently only supports Linux Platform, and to use it on other platform yo
 
 Also note that Youki currently only supports and expects systemd as init system, and would not work on other systems. There is currently work on-going to put systemd dependent features behind a feature flag, but till then you will need a systemd enabled system to work with Youki.
 
-## Requirements
+## Build Requirements
 
 As Youki is written in Rust, you will need to install and setup Rust toolchain to compile it. The instructions for that can be found on Rust's official site [here](https://www.rust-lang.org/tools/install).
+If you installed it using rustup, the correct compiler version will be setup automatically from `rust-toolchain.toml` in the repo root.
 
-You can use Youki by itself to start and run containers, but it can be a little tedious, as it is a low-level container runtime. You can use a High-level container runtime, with its runtime set to Youki, so that it will be easier to use. Both of these are explained in the [Basic Usage](./basic_usage.md). For using it along with an high-level runtime, you will to install one such as Docker or Podman. This documentation uses Docker in its examples, which can be installed from [here](https://docs.docker.com/engine/install).
+### Build with cross-rs
+
+You can compile youki using [cross-rs](https://github.com/cross-rs/cross), which provides:
+* Seamless compilation for different architectures (see `Cross.toml` in the repo root for the list of supported targets)
+* No build time dependencies (compilation runs in a container)
+* No runtime dependencies when building static binaries (musl targets)
+
+The only build dependency is [cross-rs](https://github.com/cross-rs/cross?tab=readme-ov-file#installation) and its [dependencies](https://github.com/cross-rs/cross?tab=readme-ov-file#dependencies) (rustup and docker or podman).
+
+
+```console
+$ CARGO=cross TARGET=musl just youki-dev # or youki-release
+```
+
+### Build without cross-rs
 
-To compile and run, Youki itself depends on some underlying libraries being installed. You can install them using your respective package manager as shown below.
+Install the build dependencies and then run:
+```console
+$ just youki-dev # or youki-release
+```
 
-### Debian, Ubuntu and related distributions
+Install the build depdencies using your distribution's package manger
 
+#### Debian, Ubuntu and related distributions
 ```console
 $ sudo apt-get install    \
```

Comment:
I'm not sure that we need all these dependencies.
IIUC we don't need neither of `libsystemd-dev`, `libelf-dev`, `libclang-dev` or `libssl-dev`.
Are they needed by any of the tests?

Reply:
Good point. I often forget to update the dependencies. A good way to check is to delete them from there and attempt our CI. When all CI are passed, it means they aren't necessary.
https://github.com/containers/youki/blob/656ccc3f69329c2ed10129e046d0474e6c3dee13/justfile#L153-L160